yodeploy/tests/test_application.py

```diff
@@ -36,10 +40,29 @@ class AttrDict(dict):
         store = LocalRepositoryStore(self.mkdir('artifacts'))
         self.repo = Repository(store)
         self.app = Application('test', self.tmppath('config.py'))
-        if not os.path.exists('test-data/deploy-ve/virtualenv.tar.gz'):
+
+        self.test_ve_tar_path = self._data_path('deploy-ve/virtualenv.tar.gz')
+        self.test_req_path = self._data_path('deploy-ve/requirements.txt')
+        self.test_ve_path = self._data_path('deploy-ve')
+
+        if not os.path.exists(self.test_ve_tar_path):
             self._create_test_ve()
-        self._deploy_ve_hash = ve_version(sha224sum(
-            'test-data/deploy-ve/requirements.txt'))
+        self._deploy_ve_hash = ve_version(sha224sum(self.test_req_path))
+
+    def _data_path(self, fragment):
+        version_suffix = 'py%s' % platform.python_version()
+        return os.path.join('test-data-%s' % version_suffix, fragment)
+
+    def _prep_wip_yodeploy_for_install(self):
+        """Copy the yodeploy checkout under test to a temp directory.
+
+        Returns the full path of the yodeploy copy.
+        """
```

Comment:
What are we actually concerned about here? The commit message says:

> This prevents some uncomfortable disk access to the current checkout by the easy_install command.

Which means there's something concerning, I just don't know what :)